src/IECoreGL/TextureLoader.cpp

```diff
@@ -59,24 +67,49 @@ TexturePtr TextureLoader::load( const std::string &name )
 	if( path.empty() )
 	{
 		IECore::msg( IECore::Msg::Error, "IECoreGL::TextureLoader::load", boost::format( "Couldn't find \"%s\"." ) % name );
-		m_loadedTextures[name] = nullptr; // to save us trying over and over again
+		m_loadedTextures[key] = nullptr; // to save us trying over and over again
 		return nullptr;
 	}
 
 	IECore::ReaderPtr r = IECore::Reader::create( path.string() );
 	if( !r )
 	{
 		IECore::msg( IECore::Msg::Error, "IECoreGL::TextureLoader::load", boost::format( "Couldn't create a Reader for \"%s\"." ) % path.string() );
-		m_loadedTextures[name] = nullptr; // to save us trying over and over again
+		m_loadedTextures[key] = nullptr; // to save us trying over and over again
+		return nullptr;
+	}
+
+	IECoreImage::ImageReaderPtr imageReader = IECore::runTimeCast<IECoreImage::ImageReader>( r );
```

Comment:
If we know we can only use an ImageReader, then using the generic `Reader::create()` is a bit misleading. What about replacing the conditional above with `ImageReader::canRead( fileName )`, and then just doing `new ImageReader( fileName )` here?

Reply:
How's this? 26fc68c